src/components/Visualization.tsx

```diff
@@ -233,16 +227,15 @@ const Visualization = ({ settings, toggleCatalog }: IVisualization) => {
       return;
     });
 
-    // combine steps and step edges before setting hook state
-    const combined = stepsAsElements.concat(stepEdges);
-    setElements(combined);
+    setEdges(stepEdges);
+    setNodes(stepsAsNodes);
   };
 
   // Delete an integration step
   const deleteStep = () => {
     if (!selectedStep.UUID) return;
     setIsPanelExpanded(false);
-    setSelectedStep(placeholderStep);
+    setSelectedStep({ name: '', type: '' });
```

Comment:
we can leave it, it's the same thing except instead of defining an object at the top I'm doing it inline and unnamed, only keeping the properties that are required.

Reply:
cool thanks for the clarification